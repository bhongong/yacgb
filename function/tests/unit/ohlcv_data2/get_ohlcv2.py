import ccxt
import time
#import pprint
import datetime
from datetime import timezone

#pp = pprint.PrettyPrinter(indent=4)


ex = {'kraken_ETH/USD':'pytest_ETH1__USD', 'binanceus_XLM/USD':'pytest_XL2__USD'}
tf = {'1m': 500, '1h': 500, '1d': 500}


for key in ex:
    exchange = key.split('_', 1)[0]
    market_symbol = key.split('_', 1)[1]
    #ms = ex[key].split('_', 1)[1]
    myexch = eval ('ccxt.%s ()' % exchange)
    myexch.load_markets()
    for tf_key in tf:
        timeframe = tf_key
        lim = tf[tf_key]
        print ("==", key, ex[key], tf_key, str(tf[tf_key]), "==")
        print ("Now Local Time:", datetime.datetime.now(timezone.utc))
        fticker = myexch.fetchOHLCV(market_symbol, timeframe, limit=lim)
        print ("Last OHLCV Time:", datetime.datetime.fromtimestamp(fticker[-1][0]/1000, tz=timezone.utc))
        print("")
        filename = ex[key] + '_' + timeframe + '.csv'
        with open(filename, 'w') as fl:
            for x in fticker:
                line = ''
                for y in x:
                    line += str(y) + ','
                line = line[:-1]
                print(line, file=fl)
    time.sleep(1)
