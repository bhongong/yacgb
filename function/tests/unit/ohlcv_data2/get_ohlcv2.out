== kraken_ETH/USD pytest_ETH1__USD 1m 500 ==
Now Local Time: 2021-09-10 21:46:42.232140+00:00
Last OHLCV Time: 2021-09-10 21:46:00+00:00

== kraken_ETH/USD pytest_ETH1__USD 1h 500 ==
Now Local Time: 2021-09-10 21:46:45.315840+00:00
Last OHLCV Time: 2021-09-10 21:00:00+00:00

== kraken_ETH/USD pytest_ETH1__USD 1d 500 ==
Now Local Time: 2021-09-10 21:46:48.527779+00:00
Last OHLCV Time: 2021-09-10 00:00:00+00:00

== binanceus_XLM/USD pytest_XL2__USD 1m 500 ==
Now Local Time: 2021-09-10 21:46:52.635398+00:00
Last OHLCV Time: 2021-09-10 21:46:00+00:00

== binanceus_XLM/USD pytest_XL2__USD 1h 500 ==
Now Local Time: 2021-09-10 21:46:53.014548+00:00
Last OHLCV Time: 2021-09-10 21:00:00+00:00

== binanceus_XLM/USD pytest_XL2__USD 1d 500 ==
Now Local Time: 2021-09-10 21:46:53.507846+00:00
Last OHLCV Time: 2021-09-10 00:00:00+00:00

